Site1.Master.cs

```diff
@@ -11,7 +11,45 @@ public partial class Site1 : System.Web.UI.MasterPage
     {
         protected void Page_Load(object sender, EventArgs e)
         {
+            try
+            {
+                if (Session["role"].Equals(""))
```

Comment:
Firstly, good on you for using if-else statements. Occassionally switch statements can be more organizational. 

```suggestion
             switch (Session["role"].ToString())
            {
                case A:
                    // do something here
                    break;
                case B: 
                    // do something else
                    break;
                default:
                    // do default actions
                    break;
```